src/config.cpp

```diff
@@ -0,0 +1,467 @@
+    PopulationResolver::checkDuplicatePopulations(_networkNodes);
+
+    // Load node population overrides and check biophysical types
+    _nodePopulationProperties = parser.parseNodePopulations();
+
+    // Load edge subnetowrk and check for duplicate populations
+    _networkEdges = parser.parseEdgeNetwork();
+    PopulationResolver::checkDuplicatePopulations(_networkEdges);
+
+    // Load edge population overrides
+    _edgePopulationProperties = parser.parseEdgePopulations();
+
+    const auto updateDefaultProperties =
+        [&](std::unordered_map<std::string, PopulationProperties>& map,
+            const std::string& defaultType) {
+            for (auto& entry : map) {
+                if (entry.second.type.empty())
+                    entry.second.type = defaultType;
+                if (entry.second.alternateMorphologyFormats.empty())
+                    entry.second.alternateMorphologyFormats = _components.alternateMorphologiesDir;
+                if (entry.second.biophysicalNeuronModelsDir.empty())
+                    entry.second.biophysicalNeuronModelsDir =
+                        _components.biophysicalNeuronModelsDir;
+                if (entry.second.morphologiesDir.empty())
+                    entry.second.morphologiesDir = _components.morphologiesDir;
+            }
+        };
+    updateDefaultProperties(_nodePopulationProperties, "biophysical");
+    updateDefaultProperties(_edgePopulationProperties, "chemical_synapse");
+
+    if (_components.morphologiesDir.empty())
+        PopulationResolver::checkBiophysicalPopulations(_networkNodes, _nodePopulationProperties);
+}
+
+CircuitConfig CircuitConfig::fromFile(const std::string& path) {
+    return CircuitConfig(readFile(path), fs::path(path).parent_path());
+}
+
+const std::string& CircuitConfig::getNodeSetsPath() const {
+    return _nodeSetsFile;
+}
+
+std::set<std::string> CircuitConfig::listNodePopulations() const {
+    return PopulationResolver::listPopulations(_networkNodes);
+}
+
+NodePopulation CircuitConfig::getNodePopulation(const std::string& name) const {
+    return PopulationResolver::getPopulation<NodePopulation>(name, _networkNodes);
+}
+
+std::set<std::string> CircuitConfig::listEdgePopulations() const {
+    return PopulationResolver::listPopulations(_networkEdges);
+}
+
+EdgePopulation CircuitConfig::getEdgePopulation(const std::string& name) const {
+    return PopulationResolver::getPopulation<EdgePopulation>(name, _networkEdges);
+}
+
+PopulationProperties CircuitConfig::getNodePopulationProperties(const std::string& name) const {
+    auto populations = listNodePopulations();
+    if (populations.find(name) == populations.end())
+        throw SonataError(fmt::format("Could not find node population '{}'", name));
+
+    auto popPropertiesIt = _nodePopulationProperties.find(name);
+    if (popPropertiesIt != _nodePopulationProperties.end()) {
+        return popPropertiesIt->second;
+    }
+
+    return {"biophysical",
```

Comment:
I don't think we can hardcode this; point neuron and projections are already being created.

Reply:
Do you mean to replace it by a variable to easily change the default value, or that biophysical is not anymore the default value?

Reply:
Sorry, I forgot that we had agreed on a default.  So, with some more tests, I think this is good to go.